File: frontend/src/app/core/services/venta.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { VentaResponse, VentaRequest } from '../../models/venta.model';

@Injectable({ providedIn: 'root' })
export class VentaService {

  private readonly API_URL = '/api/ventas';

  constructor(private http: HttpClient) {}

  registrar(dto: VentaRequest): Observable<VentaResponse> {
    return this.http.post<VentaResponse>(this.API_URL, dto);
  }

  obtenerPorId(id: number): Observable<VentaResponse> {
    return this.http.get<VentaResponse>(`${this.API_URL}/${id}`);
  }

  listarPorPaciente(pacienteId: number): Observable<VentaResponse[]> {
    return this.http.get<VentaResponse[]>(`${this.API_URL}/paciente/${pacienteId}`);
  }

  obtenerPorCita(citaId: number): Observable<VentaResponse | null> {
    return this.http.get<VentaResponse | null>(`${this.API_URL}/cita/${citaId}`);
  }
}

File: backend/src/main/java/com/kineticrehab/service/VentaService.java
package com.kineticrehab.service;

import com.kineticrehab.dto.request.VentaRequestDTO;
import com.kineticrehab.dto.response.VentaResponseDTO;

import java.util.List;

public interface VentaService {

    VentaResponseDTO registrar(VentaRequestDTO dto);

    VentaResponseDTO obtenerPorId(Long id);

    List<VentaResponseDTO> listarPorPaciente(Long pacienteId);

    VentaResponseDTO obtenerPorCita(Long citaId);
}
